db_sql.go

```diff
@@ -159,12 +148,23 @@ func (x *sqlUserStoreDB) CreateIdentity(email, username, firstname, lastname, mo
 	// Insert into user store
 	if tx, etx := x.db.Begin(); etx == nil {
 		// Check if the user exists (is not archived)
-		row := tx.QueryRow("SELECT email FROM authuserstore WHERE LOWER(email) = $1 AND archived = false", CanonicalizeIdentity(email))
-		scanErr := row.Scan()
-		if strings.Index(scanErr.Error(), "no rows in result set") == -1 {
-			tx.Rollback()
-			scanErr = ErrIdentityExists
-			return NullUserId, scanErr
+		if ldapUsed {
```

Comment:
Whether LDAP is sitting in front of this or not, does not belong here. That decision needs to happen before we reach sqlUserStoreDB. sqlUserStoreDB should be concerned only with it's own role. The word LDAP shouldn't appear in the methods of sqlUserStoreDB, except maybe inside a comment.
You might want to make these checks for user existence little standalone functions, to make the code more readable.